source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -102,25 +102,41 @@ def get_kraken_calls(request):
     """
     dep_modes = request['origin_mode']
     arr_modes = request['destination_mode']
+    dp_type = request.get('direct_path', 'indifferent')
+    direct_path_mode = request.get('direct_path_mode', [])
+
+    res = set()
+    if dp_type != "none":
+        for mode in direct_path_mode:
+            # to avoid duplicate tuples (mode1, mode2, "indifferent") and (mode1, mode2, "only")
+            # which will trigger 2 exact same computations
+            if mode not in dep_modes:
+                res.add((mode, mode, "only"))
 
     if len(dep_modes) == len(arr_modes) == 1:
-        return {(dep_modes[0], arr_modes[0])}
+        res.add((dep_modes[0], arr_modes[0], dp_type))
+        return res
 
     # this allowed_combinations is temporary, it does not handle all the use cases at all
     # We don't want to do ridesharing - ridesharing journeys
-    res = {c for c in fallback_modes.allowed_combinations if c in itertools.product(dep_modes, arr_modes)}
+    combinaison_mode = {
+        c for c in fallback_modes.allowed_combinations if c in itertools.product(dep_modes, arr_modes)
+    }
 
-    if not res:
+    if not combinaison_mode:
         abort(
             404,
             message='the asked first_section_mode[] ({}) and last_section_mode[] '
             '({}) combination is not yet supported'.format(dep_modes, arr_modes),
         )
 
+    for first_mode, second_mode in combinaison_mode:
```

Comment:
minor naming more accurate
```suggestion
    for dep_mode, arr_mode in combination_mode:
```